clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1766,15 +1743,15 @@ public OffsetAndMetadata committed(TopicPartition partition, final Duration time
     @Override
     public List<PartitionInfo> partitionsFor(String topic, Duration timeout) {
         acquireAndEnsureOpen();
-        long timeoutMs = timeout.toMillis();
         try {
             Cluster cluster = this.metadata.fetch();
             List<PartitionInfo> parts = cluster.partitionsForTopic(topic);
             if (!parts.isEmpty())
                 return parts;
 
+            Timer timer = time.timer(requestTimeoutMs);
```

Comment:
@hachikuji I found this issue overlooked during the review: it should be `timeout` here..